app/models/ReferenceDataPayload.scala

```diff
@@ -30,11 +30,15 @@ class ReferenceDataPayload(data: JsObject) {
 
   private lazy val lists: JsObject = (data \ "lists").get.as[JsObject]
 
-  def listsNames: collection.Set[ListName] =
-    lists.keys.map(ListName(_))
-
-  def getList(listName: ListName): SingleList =
-    SingleList(listName, messageInformation, (lists \ listName.listName \ "listEntries").as[Vector[JsObject]])
+  def toIterator(): Iterable[Seq[GenericListItem]] =
+    lists.values
+      .map {
+        list =>
+          val ln: ListName      = list.validate[ListName].get
```

Comment:
This is fairly dangerous. Why not use a `for comprehension` and a `flatMap`, I don't think we should allow the usage of `.get` here at all.

Reply:
Good point, I'm using `get` here to make it clear that in this class there are a lot of "unsafe" operations going on, but those are localised to this class, even with the `for comprehension` we'll still have a `JsLookupResult` (I think thats what we get) and we'll need to do a `get` on that. The json that this wraps needs to be schema validated before, but that hasn't been added as yet, and at that time we can do a bit of cleanup here and try to get rid of any unsafe operations